src/SymbolCollector.Server/SymbolsController.cs

```diff
@@ -325,7 +325,17 @@ public async Task ValidateBatch(Guid batchId, ModelStateDictionary modelState, C
                 && contentType == "gzip")
             {
                 await using var gzipStream = new GZipStream(section.Body, CompressionMode.Decompress);
-                await gzipStream.CopyToAsync(outputStream);
+                try
+                {
+                    await gzipStream.CopyToAsync(outputStream);
+                }
+                catch (InvalidDataException ex)
+                {
+                    _logger.LogError(ex, "Unsupported compression method encountered.");
+                    modelState.AddModelError("File", "Unsupported compression method.");
+                    code = HttpStatusCode.UnsupportedMediaType;
+                    return (null, null, code);
```

Comment:
Not sure if this is right given there's a final return at the end for failure modes

Reply:
Also CI is unhappy here, PR didn't take nullability into account:

> /Users/runner/work/symbol-collector/symbol-collector/src/SymbolCollector.Server/SymbolsController.cs(337,28): error CS8619: Nullability of reference types in value of type '(string?, Stream?, HttpStatusCode? code)' doesn't match target type '(string fileName, Stream data, HttpStatusCode? code)'.